boards/renesas/rzg3s_smarc/board.cmake

```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2024 Renesas Electronics Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+board_runner_args(jlink "--device=R9A08G045S33_M33_0" "--speed=15000")
+include(${ZEPHYR_BASE}/boards/common/jlink.board.cmake)
+
+set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
+    COMMAND ${CMAKE_OBJCOPY} -I binary -O srec --srec-forceS --adjust-vma 0x23000 ${ZEPHYR_BINARY_DIR}/${KERNEL_NAME}.bin ${ZEPHYR_BINARY_DIR}/${KERNEL_NAME}.srec
+)
```

Comment:
also quite puzzled by this, what is this doing? Why does this need to happen at a board level? Why can't this be done properly by the build system?

Reply:
Thank you for your feedback.
This will add a custom post-build step to convert the kernel binary .bin file to an S-record .srec file.
As RZ/G3S boot sequence, srec file is required for flashing. 
We plan to support more RZ boards soon, the boot sequence of other RZ boards would be different and may not require srec file so this custom post-build step shall be defined in board level.

Reply:
but Zephyr already support conversion to srec in a way which also uses the toolchain infrastructure meaning that you're not relying on a specific toolchain.

https://github.com/zephyrproject-rtos/zephyr/blob/6f26952828b037e46934a5b98d7d1d05eab45aaf/CMakeLists.txt#L1815-L1835

All you need to do is set `CONFIG_BUILD_OUTPUT_S19=y`.
Ref:
https://github.com/zephyrproject-rtos/zephyr/blob/6f26952828b037e46934a5b98d7d1d05eab45aaf/Kconfig.zephyr#L749

So why are you not using existing functionality ?

Reply:
Thank you for providing this information.
We tried to use it, but it only generates `*.s19` file. The bootloader only support srec file format.
Are there any other options or configurations to enable srec generation within the build system?
If no, we should consider updating [CMakeLists.txt](https://github.com/zephyrproject-rtos/zephyr/blob/6f26952828b037e46934a5b98d7d1d05eab45aaf/CMakeLists.txt#L1815-L1835) to support srec generation.

Reply:
Sorry, I misunderstood that s19 is a different format from srec.
I updated as your suggestion. 
Please kindly check it.